dkist/wcs/models.py

```diff
@@ -394,14 +676,60 @@ def _calculate_separability_matrix(self):
         cleft[: sepleft.shape[0], : sepleft.shape[1]] = sepleft
 
         cright = np.zeros((noutp, sepright.shape[1]))
-        cright[-sepright.shape[0]:, -sepright.shape[1]:] = 1
+        cright[-sepright.shape[0]:, -sepright.shape[1]:] = sepright
 
         left_solo_end = self.left.n_inputs - self.shared_inputs
         right_solo_start = left_solo_end + self.shared_inputs
         matrix = np.zeros((self.n_outputs, self.n_inputs))
         matrix[:, :left_solo_end] = cleft[:, :left_solo_end]
         matrix[:, left_solo_end:right_solo_start] = np.logical_or(cleft[:, left_solo_end:],
                                                                   cright[:, :self.shared_inputs])
-        matrix[:, right_solo_start:] = cright[:, :right_solo_start]
+        matrix[:, right_solo_start:] = cright[:, self.shared_inputs:]
 
         return matrix
+
+
+def varying_celestial_transform_from_tables(crpix: Union[Iterable[float], u.Quantity],
```

Comment:
Why `Iterable[float]`, but not `Iterable[u.Quantity]`?

Reply:
Because Quantity is a numpy array so already iterable.